.github/workflows/ci.yml

```diff
@@ -11,8 +11,14 @@ jobs:
         env:
           token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Does this need a token for just npm tooling ? Could remove this as before we enable actions.

Reply:
Unfortunately, it does need the token. To show what happens when the token is missing, I created commit 333a4216753aa4af9d19ea37ced7da23c98cd5b7 and ran it through the CI pipeline: https://github.com/striezel-stash/rustsec-audit-check/actions/runs/4433071828/jobs/7777762429

It fails when trying to install the dependencies via `npm ci`:
```
npm ERR! code E401
npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/download/@actions-rs/core/0.0.9/5dcce7d4097ba352a07b52206b4956d6511db0e4171473fa10a92b884dd09e02 - authentication token not provided

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-03-16T02_54_03_553Z-debug-0.log
Error: Process completed with exit code 1.
```
Basic problem here is that `@actions-rs/core` is not hosted at the regular NPM registry (registry.npmjs.org), but at npm.pkg.github.com instead, and that registry needs a token for authorization of the package download.

So unless somebody finds a replacement for [`@actions-rs/core`](https://github.com/actions-rs/core) or manages to host it somewhere else, the token is required.

Reply:
Could we re-base this ? I've merged #6 that gets rid of the token req via using setup-node action.

Reply:
Yes, we could. I've applied the changes from #6 in commit a7604c78f5ab8e961f94df5f7e0a7f5ec5ac7eae on a separate branch. However, the changes of PR #6 do not have the effect you might think they have, and the build fails with the rebased changes from #6: https://github.com/striezel-stash/rustsec-audit-check/actions/runs/4526586890/jobs/7971860636.

The action `action/setup-node` does not remove the requirement for authentication. `action/setup-node` only installs a certain version of Node.js, i. e. the runtime. It does _not_ install any packages from the npmjs registry or any other registry as `npm install` or `npm ci` commands do, and it does not circumvent the required authorization. These are completely orthogonal tasks.

Therefore, the changes from #6 should be reverted, before this PR here is merged.

Reply:
I forked the actions-rs/core to rinse-repeat/actions-rs-core and now we don't need the token.
Since I didn't have permission to push to this branch I superceded this PR and I marked you as co-authored from these commits to the superceded PR.
It still needs a lot of work but anyways.
Cheers